### DISCUSSION TITLE
Why the query IDs have to be integers

### DISCUSSION THREAD

**Discussion by kwang2049** [castorini/pyserini#1003] [Q&A] [answered]
Hi,

I saw at [this line](https://github.com/castorini/pyserini/blob/8b37bb31d24c42b36887b6a6a48d4a17c877e4b6/pyserini/search/lucene/_base.py#L288), pyserini needs to make sure the query IDs are integers. May I ask why does it need that? It would be very inconvenient for some datasets and one need to re-map the query IDs from strings to integers somehow.

**Comment by lintool** [comment, marked as answer]
They don't need to be integers. E.g.,

```python
pyserini.search.lucene.get_topics('car17v1.5-benchmarkY1test').keys()
```

The topic ids are:

```
dict_keys(['Drinking%20water/Regulation/United%20States', 'Beach%20nourishment/Alternatives/complements%20to%20nourishment', 
...
```
